tools/get-prebuilt.js

```diff
@@ -1,71 +1,111 @@
+                return;
+            }
+
+            const content = [];
+            res
+                .on("data", (chunk) => {
+                    content.push(chunk);
+                })
+                .on("end", () => {
+                    const buffer = Buffer.concat(content);
+                    fs.writeFileSync(romPath, buffer);
+                    rom.urls = romPath;
+                    process.stdout.write("... Done\n");
+                    resolve(rom);
+                })
+                .on('error', (err) => {
+                    reject(err);
+                });
+        }).on('error', (e) => {
+            console.log(`An error occurred downloading ${url}`);
+            reject(e);
         });
     });
-  });
 };
 
-(async () => {
-  const data = await fetch(prebuilt_json_url);
-  json = await data.json();
+const getAll = async () => {
+    const data = await fetch(prebuilt_json_url).catch((err) => {
+        console.log(`An error occurred downloading ${prebuilt_json_url}`);
+        console.error(err);
+        process.exit(1);
+    });
+    json = await data.json();
 
-  const indexFile = {
-    indexversion: json.indexversion,
-    latest: null,
-    nightly: [],
-    stable: [],
-  };
+    const indexFile = {
+        indexversion: json.indexversion,
+        latest: null,
+        nightly: [],
+        stable: [],
+    };
 
-  if (json.latest) {
-    indexFile.latest = await getRom(json.latest).catch((err) => {
-      return json.latest;
-    });
-  }
-  if (json.nightly?.length > 0) {
-    for (let i = 0; i < json.nightly.length; i++) {
-      indexFile.nightly.push(await getRom(json.nightly[i]).catch((err) => {
-        return json.nightly[i];
-      }));
+    console.log('\nLatest:');
+    if (json.latest) {
+        indexFile.latest = await getRom(json.latest).catch((err) => {
+            return null;
+        });
     }
-  }
-  if (json.stable?.length > 0) {
-    for (let i = 0; i < json.stable.length; i++) {
-      indexFile.stable.push(await getRom(json.stable[i]).catch((err) => {
-        return json.stable[i];
-      }));
+    console.log('\nNightly:');
+    if (json.nightly?.length > 0) {
+        for (let i = 0; i < json.nightly.length; i++) {
+            let rom = await getRom(json.nightly[i]).catch((err) => {
+                return null;
+            })
+            if(rom) {
+                indexFile.nightly.push(rom);
+            }
+        }
     }
-  }
+    console.log('\nStable:');
+    if (json.stable?.length > 0) {
+        for (let i = 0; i < json.stable.length; i++) {
+            let rom = await getRom(json.stable[i]).catch((err) => {
+                return null;
+            });
+            if(rom) {
+                indexFile.stable.push(rom);
+            }
+        }
+    }
+
+    console.log('\n\nWriting roms/index.json');
+    fileString = JSON.stringify(indexFile, false, 2);
+    fs.writeFileSync(`roms/index.json`, fileString);
+};
+
+getAll();
```

Comment:
this should be wrapped in an IIFE - to limit its scope, and we should await it's return as it's making async calls.



```suggestion
(async () => {
  console.log("Getting Pre-built ROMs");
  awaut getAll();
})(); 
```